.circleci/config.yml

```diff
@@ -237,6 +260,48 @@ commands:
             gsutil cp  ${TAR_PATH} "gs://${BUILDS_BUCKET}/${ANGULAR_PROJECT_NAME}/${TAR_NAME}"
           working_directory: ~/repo
 
+  conditionally-launch-build-and-deploy:
+    description: Launch build and deploy if project has changed
+    parameters:
+      study_keys:
+        type: string
+      study_guids:
+        type: string
+    steps:
+      - run:
+          name: Check << parameters.study_keys >> and launch build if modified
+          command: |
+            set -u
+            set +e
+            set +x
+            # CircleCi does not support arrays as params, so here is my workaround
+            STUDY_KEYS=(<< parameters.study_keys >>)
+            STUDY_GUIDS=(<< parameters.study_guids >>)
+
+            CHANGED_PROJECTS=$(findmodifiedprojects.sh <<pipeline.git.base_revision >> << pipeline.git.revision >>)
+
+            for i in "${!STUDY_KEYS[@]}"; do
+              STUDY_KEY=${STUDY_KEYS[$i]}
+              STUDY_GUID=${STUDY_GUIDS[$i]}
+
+              NG_PROJECT_NAME="ddp-$STUDY_KEY"
+              echo  $CHANGED_PROJECTS |
+                  grep -E -e $NG_PROJECT_NAME -e _SHARED_  -e ddp-sdk -e toolkit
+              if [[ $? -eq 0 ]]; then
```

Comment:
If the changed projects included one of the grep expressions, we are going to initiate a build on the current study in the loop